apps/src/p5lab/layoutUtils.js

```diff
@@ -0,0 +1,183 @@
+
+  const minX = 20;
+  const maxX = 400 - minX;
+  const minY = 35;
+  const maxY = 400 - 40;
+  const radiansToDegrees = 180 / Math.PI;
+  const maxCircleRadius = 165;
+
+  if (layout === 'top') {
+    createRow(group, 100);
+  } else if (layout === 'row') {
+    createRow(group, 200);
+  } else if (layout === 'bottom') {
+    createRow(group, 300);
+  } else if (layout === 'border') {
+    // First fill the four corners.
+    // Then split remainder into 4 groups. Distribute group one along the top,
+    // group 2 along the right, etc.
+    if (count > 0) {
+      group[0].x = minX;
+      group[0].y = minY;
+      group[0].rotation = 0;
+    }
+    if (count > 1) {
+      group[1].x = maxX;
+      group[1].y = minY;
+      group[1].rotation = 0;
+    }
+    if (count > 2) {
+      group[2].x = maxX;
+      group[2].y = maxY;
+      group[2].rotation = 0;
+    }
+    if (count > 3) {
+      group[3].x = minX;
+      group[3].y = maxY;
+      group[3].rotation = 0;
+    }
+    if (count > 4) {
+      const topCount = Math.ceil((count - 4 - 0) / 4);
+      const rightCount = Math.ceil((count - 4 - 1) / 4);
+      const bottomCount = Math.ceil((count - 4 - 2) / 4);
+      const leftCount = Math.ceil((count - 4 - 3) / 4);
+
+      for (let i = 0; i < topCount; i++) {
+        const sprite = group[4 + i];
+        // We want to include the corners in our total count so that the first
+        // inner sprite is > 0 and the last inner sprite is < 1 when we lerp.
+        sprite.x = p5.lerp(minX, maxX, (i + 1) / (topCount + 1));
+        sprite.y = minY;
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < rightCount; i++) {
+        const sprite = group[4 + topCount + i];
+        sprite.x = maxX;
+        sprite.y = p5.lerp(minY, maxY, (i + 1) / (rightCount + 1));
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < bottomCount; i++) {
+        const sprite = group[4 + topCount + rightCount + i];
+        sprite.x = p5.lerp(minX, maxX, (i + 1) / (bottomCount + 1));
+        sprite.y = maxY;
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < leftCount; i++) {
+        const sprite = group[4 + topCount + rightCount + bottomCount + i];
+        sprite.x = minX;
+        sprite.y = p5.lerp(minY, maxY, (i + 1) / (leftCount + 1));
+        sprite.rotation = 0;
+      }
+    }
+  } else if (layout === 'circle') {
+    // Adjust radius of circle and size of the sprite according to number of
+    // sprites in our group.
+    const pct = p5.constrain(count / 10, 0, 1);
+    const radius = p5.lerp(50, maxCircleRadius, pct);
+    const scale = p5.lerp(0.8, 0.3, pct * pct);
+    const startAngle = -Math.PI / 2;
+    const deltaAngle = (2 * Math.PI) / count;
+
+    group.forEach((sprite, i) => {
+      const angle = deltaAngle * i + startAngle;
+      sprite.x = 200 + radius * Math.cos(angle);
+      sprite.y = 200 + radius * Math.sin(angle);
+      sprite.rotation = (angle - startAngle) * radiansToDegrees;
```

Comment:
After playing with this locally, I don't believe we want to adjust the rotation of the sprites if we're in Sprite Lab. Dance Party's sprites are all anthropomorphic/humanoid. Students don't actually program the sprites to move around in the same way they do in Sprite Lab. Messing with the rotation could create some false impressions of how movement will be impacted.

Here's a comparison of how it works right now, with how it would work if we kept the rotation at 0. Note that after creating the two groups, each is told to move a different direction using a behavior.

![rotated circle sprites](https://user-images.githubusercontent.com/43474485/230111382-f0245dca-911e-4ba8-9832-5851cbef977c.gif)
![not rotated](https://user-images.githubusercontent.com/43474485/230111403-a56d3fb5-9aef-47d9-b582-07e5c8a5a19d.gif)

Reply:
Going a step further, I think that for Sprite Lab, we don't want to even set the rotation to 0 and instead just leave it what it was originally. This won't affect sprites that were just created, but if the blue layout block is used in conjunction with events, re-initializing the sprite rotation leads to undesired  out comes if the sprites are moving around and changing directions.

In this example, note that the rocket ships seem to be able to be turned and driven forward using the arrow keys, but only until the layout is changed into a circle. Once that happens, the movement direction is messed up.


https://user-images.githubusercontent.com/43474485/230114097-ca4faafb-1171-4dc1-842f-9be98f136571.mp4